[develop]: Add linkcheck & RTD theme; re-organize documentation (#263)
hafs-community/HAFS: doc/UsersGuide/source/CustomizingTheWorkflow/CDEPS.rst

By default, the DAPP keyword should already be set to HAFS-ALL on all supported machines except wcoss_cray.
 

> The remainder of the build process is the same as described in the HAFS installation guide.

Reviewer: ```suggestion
The remainder of the build process is the same as described in the :ref:`HAFS Quick Start Guide <QuickStart>`.
```

.. _fig_datm:

> .. figure:: https://raw.githubusercontent.com/wiki/hafs-community/HAFS/docs_images/hafs_cdeps_workflow_docn.png

Reviewer: Revert links in this chapter, too.
I believe the pic should still be for DATM, not DOCN.

> .. figure:: https://raw.githubusercontent.com/wiki/hafs-community/HAFS/docs_images/hafs_cdeps_workflow_docn.png

Reviewer: ```suggestion
.. figure:: https://github.com/hafs-community/HAFS/wiki/docs_images/hafs_cdeps_workflow_datm.png
```

    :width: 50 %
    :alt: HAFS-CDEPS workflow for DATM

    *Schematic diagram of the HAFS-CDEPS workflow for DATM. Blue text indicates the jobs that will run. Gray text indicates jobs that only run when data models are not used.*

.. _fig_docn:

> .. figure:: https://raw.githubusercontent.com/wiki/hafs-community/HAFS/docs_images/hafs_cdeps_workflow_docn.png

Reviewer: ```suggestion
.. figure:: https://github.com/hafs-community/HAFS/wiki/docs_images/hafs_cdeps_workflow_docn.png
```

.. _fig_docn:

> .. figure:: https://github.com/hafs-community/HAFS/wiki/docs_images/hafs_cdeps_workflow_datm.png

Reviewer: Somehow when the other one got changed to datm, this one did too. 
```suggestion
.. figure:: https://github.com/hafs-community/HAFS/wiki/docs_images/hafs_cdeps_workflow_docn.png
```
